src/invite-api.js

```diff
@@ -1,236 +1,440 @@
 // @ts-check
 import { TypedEmitter } from 'tiny-typed-emitter'
 import { InviteResponse_Decision } from './generated/rpc.js'
-import { projectKeyToId, projectKeyToPublicId } from './utils.js'
+import { assert, keyToId } from './utils.js'
+import timingSafeEqual from './lib/timing-safe-equal.js'
+import { promiseWithResolvers } from './ponyfills.js'
 
 /**
- * @typedef {Object} InviteApiEvents
+ * Internally, we typically use the `Invite` type from the protobuf. We also use
+ * an external type for public consumers.
```

Comment:
I don't love this, but it's the best of three options I evaluated. The main issue: we want to use `Buffer`s over the wire, but `string`s in the public API.

The three options I evaluated:

1. Always use strings, even in the protos
2. Always use strings _except_ in the protos; we'd convert to/from buffers when sending/receiving the bytes
3. Only use strings for the public API, and use `Buffer`s everywhere else

Better ideas are welcome!